examples/foo.complex.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2012-2015 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import shutil
+import time
+import random
+
+import luigi
+
+max_depth = 10
+max_total_nodes = 50
+current_nodes = 0
+
+
+class Foo(luigi.Task):
+
+    def run(self):
+        print "Running Foo"
+
+    def requires(self):
+        global current_nodes
+        for i in xrange(30 / max_depth):
+            current_nodes += 1
+            yield Bar(i)
+
+
+class Bar(luigi.Task):
```

Comment:
max_depth, max_total_nodes, current_nodes are global variables, to store the STATE of the graph that is going to get built. The idea is that you can specify by parameters the size and limits of the foo graph. This means that this variables are needed globally within each luigi's task (requires and run) so the graph generation has a stop condition, while been dynamically generated on each run.